Lib/glyphsLib/builder/sources.py

```diff
@@ -151,30 +151,23 @@ def _to_designspace_source_layer(self):
 
     # First, collect all brace layers in the font and which glyphs and which masters
     # they belong to.
-    layer_name_to_master_ids = collections.defaultdict(set)
-    layer_name_to_glyph_names = collections.defaultdict(list)
+    brace_coordinates_to_master_ids = collections.defaultdict(set)
+    brace_coordinates_to_glyph_names = collections.defaultdict(list)
     for glyph in self.font.glyphs:
         for layer in glyph.layers:
-            if (
-                "{" in layer.name
-                and "}" in layer.name
-                and ".background" not in layer.name
-            ):
-                layer_name_to_master_ids[layer.name].add(layer.associatedMasterId)
-                layer_name_to_glyph_names[layer.name].append(glyph.name)
+            if layer._is_brace_layer():
+                brace_coordinates = tuple(layer._brace_coordinates())
+                brace_coordinates_to_master_ids[brace_coordinates].add(
+                    layer.associatedMasterId
+                )
+                brace_coordinates_to_glyph_names[brace_coordinates].append(glyph.name)
 
     # Next, insert the brace layers in a defined location in the existing designspace.
     designspace = self._designspace
     layers_to_insert = collections.defaultdict(list)
-    for layer_name, master_ids in layer_name_to_master_ids.items():
-        # Construct coordinates first...
-        brace_coordinates = [
-            float(c)
-            for c in layer_name[
-                layer_name.index("{") + 1 : layer_name.index("}")
-            ].split(",")
-        ]
-
+    for key, master_ids in brace_coordinates_to_master_ids.items():
+        brace_coordinates = list(key)
+        layer_name = "{" + ", ".join(f"{c:g}" for c in brace_coordinates) + "}"
```

Comment:
hm so your're now discarding the original layer name.. why? I think we should try to retain the original brace layer names.

Reply:
I would not use the layer internally any more. Wouldn't it be easier to use the attributes and converting layer names to attributes for v2 files?

Reply:
> I would not use the layer internally any more. Wouldn't it be easier to use the attributes and converting layer names to attributes for v2 files?

This is what I wanted to do originally, but format version is not always known when layer name is set, so we can't set the attributes there, and I don't know how to make attributes some kind of dynamic dictionary that would check layer name when an attribute is accessed (and whether or not this would have performance imprecations).

Reply:
> hm so your're now discarding the original layer name.. why? I think we should try to retain the original brace layer names.

This is to make the tests pass with V3 file, since one of the tests is checking design space layer name and using V3 layer name makes no sense as it can be anything.

Reply:
This is also inline with Glyphs 3, where the displayed layer name is always constructed for layers with attributes.

Reply:
I have a read only property `nameUI` for that.

Reply:
> inline with Glyphs 3, where the displayed layer name is always constructed for layers with attributes.

I see, but for Glyphs 2 brace layer names could contain some other string that added context to the numbers in the parentheses. I'd still consider it a (minor?) regression to lose that.

Reply:
I don’t mind reverting this part, but I’ll need a way to make `test_designspace_generation_brace_layers` test pass without it.

Reply:
Actually this change is brooken, since now designspace layer names don’t actually match UFO layer names. Let me try to fix it.

Reply:
Should be fixed now (I hop, I this the use of layer names in V3 files is wrong, but fixing this probably is a bigger change than this place.